module/zfs/zil.c

```diff
@@ -1560,10 +1639,10 @@ zil_lwb_write_done(zio_t *zio)
 	nlwb = list_next(&zilog->zl_lwb_list, lwb);
 	if (nlwb && nlwb->lwb_state != LWB_STATE_ISSUED)
 		nlwb = NULL;
-	mutex_exit(&zilog->zl_lock);
```

Comment:
Is my fix above somehow insufficient that you are extending zl_lock and using lwb_vdev_lock below?  I suspect what are doing is expensive, unneeded and would not solve the problem any way.

Reply:
Ah, this patch predates your fix, so it may not work correctly with your changes in place. It definitely fixed the issue I described in the PR notes, though.